src/connection.rs

```diff
@@ -428,8 +428,9 @@ impl Connection {
 
         let client_ref = &client;
 
-        let _ret = client
-            .sync_with_callback(sync_settings, |response| async move {
+        loop {
+            let ret = client
+            .sync_with_callback(sync_settings.clone(), |response| async move {
```

Comment:
There's nowadays a `SyncServic` which handles this looping automatically.

I think it'll make sense to eventually migrate over to it: https://docs.rs/matrix-sdk-ui/latest/matrix_sdk_ui/sync_service/struct.SyncService.html#example.

Reply:
I've seen the matrix_sdk_ui crate but I haven't explored it (yet, also it's been so long that I was used to just matrix_sdk). I'll try to experiment with it at some point.

Reply:
It builds on top of the `matrix-sdk` crate and provides some higher level and mostly UI client specific functionality.